### ISSUE TITLE
bindgen tests fail on 32-bit architectures (i686 and armv7)

### ISSUE THREAD

**Issue by decathorpe** [servo/rust-freetype#64]
It looks like there are struct layouts whose size is architecture-dependent and that trip up bindgen (or at least the bindgen tests).

This is easily reproduced on a x86 / 32 bit (i686) or armv7 (armv7hl) machine by just running `cargo test` (confirmed against git master branch) - log output below.

Can those failures be ignored or are they indicative of a real issue on 32 bit arches?

```
$ cargo test
(...)
running 28 tests
test freetype::bindgen_test_layout_FT_Bitmap_ ... FAILED
test freetype::bindgen_test_layout_FT_BBox_ ... FAILED
test freetype::bindgen_test_layout_FT_Bitmap_Size_ ... FAILED
test freetype::bindgen_test_layout_FT_Data_ ... FAILED
test freetype::bindgen_test_layout_FT_CharMapRec_ ... FAILED
test freetype::bindgen_test_layout_FT_Generic_ ... FAILED
test freetype::bindgen_test_layout_FT_GlyphSlotRec_ ... FAILED
test freetype::bindgen_test_layout_FT_Glyph_Metrics_ ... FAILED
test freetype::bindgen_test_layout_FT_ListRec_ ... FAILED
test freetype::bindgen_test_layout_FT_ListNodeRec_ ... FAILED
test freetype::bindgen_test_layout_FT_Matrix_ ... FAILED
test freetype::bindgen_test_layout_FT_FaceRec_ ... FAILED
test freetype::bindgen_test_layout_FT_MemoryRec_ ... FAILED
test freetype::bindgen_test_layout_FT_Open_Args_ ... FAILED
test freetype::bindgen_test_layout_FT_Module_Class_ ... FAILED
test freetype::bindgen_test_layout_FT_Outline_Funcs_ ... FAILED
test freetype::bindgen_test_layout_FT_Outline_ ... FAILED
test freetype::bindgen_test_layout_FT_SizeRec_ ... FAILED
test freetype::bindgen_test_layout_FT_Parameter_ ... FAILED
test freetype::bindgen_test_layout_FT_Raster_Params_ ... FAILED
test freetype::bindgen_test_layout_FT_Span_ ... ok
test freetype::bindgen_test_layout_FT_Raster_Funcs_ ... FAILED
test freetype::bindgen_test_layout_FT_Size_Metrics_ ... FAILED
test freetype::bindgen_test_layout_FT_UnitVector_ ... ok
test freetype::bindgen_test_layout_FT_Size_RequestRec_ ... FAILED
test freetype::bindgen_test_layout_FT_StreamDesc_ ... FAILED
test freetype::bindgen_test_layout_FT_StreamRec_ ... FAILED
test freetype::bindgen_test_layout_FT_Vector_ ... FAILED
failures:
---- freetype::bindgen_test_layout_FT_Bitmap_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Bitmap_' panicked at 'assertion failed: `(left == right)`
  left: `24`,
 right: `40`: Size of: FT_Bitmap_', src/freetype.rs:392:5
---- freetype::bindgen_test_layout_FT_BBox_ stdout ----
thread 'freetype::bindgen_test_layout_FT_BBox_' panicked at 'assertion failed: `(left == right)`
  left: `16`,
 right: `32`: Size of: FT_BBox_', src/freetype.rs:335:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
---- freetype::bindgen_test_layout_FT_Bitmap_Size_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Bitmap_Size_' panicked at 'assertion failed: `(left == right)`
  left: `16`,
 right: `32`: Size of: FT_Bitmap_Size_', src/freetype.rs:1359:5
---- freetype::bindgen_test_layout_FT_Data_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Data_' panicked at 'assertion failed: `(left == right)`
  left: `8`,
 right: `16`: Size of: FT_Data_', src/freetype.rs:870:5
---- freetype::bindgen_test_layout_FT_CharMapRec_ stdout ----
thread 'freetype::bindgen_test_layout_FT_CharMapRec_' panicked at 'assertion failed: `(left == right)`
  left: `12`,
 right: `16`: Size of: FT_CharMapRec_', src/freetype.rs:1814:5
---- freetype::bindgen_test_layout_FT_Generic_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Generic_' panicked at 'assertion failed: `(left == right)`
  left: `8`,
 right: `16`: Size of: FT_Generic_', src/freetype.rs:900:5
---- freetype::bindgen_test_layout_FT_GlyphSlotRec_ stdout ----
thread 'freetype::bindgen_test_layout_FT_GlyphSlotRec_' panicked at 'assertion failed: `(left == right)`
  left: `160`,
 right: `304`: Size of: FT_GlyphSlotRec_', src/freetype.rs:1685:5
---- freetype::bindgen_test_layout_FT_Glyph_Metrics_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Glyph_Metrics_' panicked at 'assertion failed: `(left == right)`
  left: `32`,
 right: `64`: Size of: FT_Glyph_Metrics_', src/freetype.rs:1299:5
---- freetype::bindgen_test_layout_FT_ListRec_ stdout ----
thread 'freetype::bindgen_test_layout_FT_ListRec_' panicked at 'assertion failed: `(left == right)`
  left: `8`,
 right: `16`: Size of: FT_ListRec_', src/freetype.rs:960:5
---- freetype::bindgen_test_layout_FT_ListNodeRec_ stdout ----
thread 'freetype::bindgen_test_layout_FT_ListNodeRec_' panicked at 'assertion failed: `(left == right)`
  left: `12`,
 right: `24`: Size of: FT_ListNodeRec_', src/freetype.rs:928:5
---- freetype::bindgen_test_layout_FT_Matrix_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Matrix_' panicked at 'assertion failed: `(left == right)`
  left: `16`,
 right: `32`: Size of: FT_Matrix_', src/freetype.rs:833:5
---- freetype::bindgen_test_layout_FT_FaceRec_ stdout ----
thread 'freetype::bindgen_test_layout_FT_FaceRec_' panicked at 'assertion failed: `(left == right)`
  left: `132`,
 right: `248`: Size of: FT_FaceRec_', src/freetype.rs:1454:5
---- freetype::bindgen_test_layout_FT_MemoryRec_ stdout ----
thread 'freetype::bindgen_test_layout_FT_MemoryRec_' panicked at 'assertion failed: `(left == right)`
  left: `16`,
 right: `32`: Size of: FT_MemoryRec_', src/freetype.rs:137:5
---- freetype::bindgen_test_layout_FT_Open_Args_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Open_Args_' panicked at 'assertion failed: `(left == right)`
  left: `32`,
 right: `64`: Size of: FT_Open_Args_', src/freetype.rs:2010:5
---- freetype::bindgen_test_layout_FT_Module_Class_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Module_Class_' panicked at 'assertion failed: `(left == right)`
  left: `36`,
 right: `72`: Size of: FT_Module_Class_', src/freetype.rs:2381:5
---- freetype::bindgen_test_layout_FT_Outline_Funcs_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Outline_Funcs_' panicked at 'assertion failed: `(left == right)`
  left: `24`,
 right: `48`: Size of: FT_Outline_Funcs_', src/freetype.rs:527:5
---- freetype::bindgen_test_layout_FT_Outline_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Outline_' panicked at 'assertion failed: `(left == right)`
  left: `20`,
 right: `40`: Size of: FT_Outline_', src/freetype.rs:453:5
---- freetype::bindgen_test_layout_FT_SizeRec_ stdout ----
thread 'freetype::bindgen_test_layout_FT_SizeRec_' panicked at 'assertion failed: `(left == right)`
  left: `44`,
 right: `88`: Size of: FT_SizeRec_', src/freetype.rs:1628:5
---- freetype::bindgen_test_layout_FT_Parameter_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Parameter_' panicked at 'assertion failed: `(left == right)`
  left: `8`,
 right: `16`: Size of: FT_Parameter_', src/freetype.rs:1977:5
---- freetype::bindgen_test_layout_FT_Raster_Params_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Raster_Params_' panicked at 'assertion failed: `(left == right)`
  left: `48`,
 right: `96`: Size of: FT_Raster_Params_', src/freetype.rs:647:5
---- freetype::bindgen_test_layout_FT_Raster_Funcs_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Raster_Funcs_' panicked at 'assertion failed: `(left == right)`
  left: `24`,
 right: `48`: Size of: FT_Raster_Funcs_', src/freetype.rs:737:5
---- freetype::bindgen_test_layout_FT_Size_Metrics_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Size_Metrics_' panicked at 'assertion failed: `(left == right)`
  left: `28`,
 right: `56`: Size of: FT_Size_Metrics_', src/freetype.rs:1900:5
---- freetype::bindgen_test_layout_FT_Size_RequestRec_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Size_RequestRec_' panicked at 'assertion failed: `(left == right)`
  left: `20`,
 right: `32`: Size of: FT_Size_RequestRec_', src/freetype.rs:2113:5
---- freetype::bindgen_test_layout_FT_StreamDesc_ stdout ----
thread 'freetype::bindgen_test_layout_FT_StreamDesc_' panicked at 'assertion failed: `(left == right)`
  left: `4`,
 right: `8`: Alignment of FT_StreamDesc_', src/freetype.rs:269:5
---- freetype::bindgen_test_layout_FT_StreamRec_ stdout ----
thread 'freetype::bindgen_test_layout_FT_StreamRec_' panicked at 'assertion failed: `(left == right)`
  left: `48`,
 right: `80`: Size of: FT_StreamRec_', src/freetype.rs:199:5
---- freetype::bindgen_test_layout_FT_Vector_ stdout ----
thread 'freetype::bindgen_test_layout_FT_Vector_' panicked at 'assertion failed: `(left == right)`
  left: `8`,
 right: `16`: Size of: FT_Vector_', src/freetype.rs:306:5
failures:
    freetype::bindgen_test_layout_FT_BBox_
    freetype::bindgen_test_layout_FT_Bitmap_
    freetype::bindgen_test_layout_FT_Bitmap_Size_
    freetype::bindgen_test_layout_FT_CharMapRec_
    freetype::bindgen_test_layout_FT_Data_
    freetype::bindgen_test_layout_FT_FaceRec_
    freetype::bindgen_test_layout_FT_Generic_
    freetype::bindgen_test_layout_FT_GlyphSlotRec_
    freetype::bindgen_test_layout_FT_Glyph_Metrics_
    freetype::bindgen_test_layout_FT_ListNodeRec_
    freetype::bindgen_test_layout_FT_ListRec_
    freetype::bindgen_test_layout_FT_Matrix_
    freetype::bindgen_test_layout_FT_MemoryRec_
    freetype::bindgen_test_layout_FT_Module_Class_
    freetype::bindgen_test_layout_FT_Open_Args_
    freetype::bindgen_test_layout_FT_Outline_
    freetype::bindgen_test_layout_FT_Outline_Funcs_
    freetype::bindgen_test_layout_FT_Parameter_
    freetype::bindgen_test_layout_FT_Raster_Funcs_
    freetype::bindgen_test_layout_FT_Raster_Params_
    freetype::bindgen_test_layout_FT_SizeRec_
    freetype::bindgen_test_layout_FT_Size_Metrics_
    freetype::bindgen_test_layout_FT_Size_RequestRec_
    freetype::bindgen_test_layout_FT_StreamDesc_
    freetype::bindgen_test_layout_FT_StreamRec_
    freetype::bindgen_test_layout_FT_Vector_
test result: FAILED. 2 passed; 26 failed; 0 ignored; 0 measured; 0 filtered out
```

**Comment by jdm** [comment]
I suspect they are ignorable failures - the cases I checked all involve checking the size of structs that contain pointer values. If we ran bindgen as part of the build instead of checking in the generated file, the tests would have the expected 32-bit sizes and would pass. Since the underlying types are using Rust types that are appropriately sized on both 32 and 64-bit architectures, I believe that using this crate should be safe.
